src/vm/node_modules/VM.js

```diff
@@ -10897,12 +10897,29 @@ exports.reprovision = function (uuid, payload, options, callback)
             vs.once('ready', function (ready_ev) {
                 try {
                     vmobj = vminfodEventGetVM(uuid, ready_ev);
+
+                    if (!vmobj.zpool) {
+                        log.warn('failed to load \"zpool\" field, retrying '
+                            + 'via manual load');
+
+                        VM.load(uuid, {
+                            log: log,
+                            loadManually: true
+                        }, function (err, obj) {
+                            if (err) {
+                                cb(err);
+                                return;
```

Comment:
What do we do with "vmobj" in this case of failure? Feels like we'll leak something here.